### ISSUE TITLE
Add bouncer API support in measurement-kit

### ISSUE THREAD

**Issue by hellais** [measurement-kit/measurement-kit#693]
Currently measurement-kit uses a set of hardcoded values for collectors and test-helpers.

Ideally we should support communicating with the canonical ooni-probe bouncer and obtain the test-helpers and collector addresses from it.

**Comment by bassosimone** [comment]
We can start implementing this by implementing an HTTP client for communicating with a specific bouncer for a specific test. For now, I'll just focus on the minimum amount of information we need to start off:

```C++
namespace mk {
namespace ooni {
namespace bouncer {

struct Reply {
    std::string https_collector;
    std::string https_helper;
};

void post_net_tests(
    std::string base_bouncer_url, // Example: "https://a.collector.ooni.io/bouncer/net-tests"
    std::string test_name,             // Example: "web_connectivity"
    std::string test_version,          // Example: "0.0.1"
    std::list<std::string> helpers,  // Example: ["web-connectivity"]
    Callback<Error, Reply> cb,
    Settings settings,
    Reactor reactor,
    Logger logger
);

}}}
```

Example JSON request to be sent for web_connectivity:

```JSON
{
    "net-tests": [
        {
            "input-hashes": [],
            "name": "web_connectivity",
            "test-helpers": [
                "web-connectivity"
            ],
            "version": "0.0.1"
        }
    ]
}
```

The method shall be `POST` and the request url path `/bouncer/net-tests`. Assuming `XO` contains the above JSON, with curl this would be:

```
curl -vd @XO https://a.collector.ooni.io/bouncer/net-tests
```

The JSON response, from which we should parse enough information to fill the structure above, looks like this:

```JSON
{
    "net-tests": [
        {
            "collector": "httpo://ihiderha53f36lsd.onion",
            "collector-alternate": [
                {
                    "address": "https://a.collector.ooni.io:4441",
                    "type": "https"
                },
                {
                    "address": "https://das0y2z2ribx3.cloudfront.net",
                    "front": "a0.awsstatic.com",
                    "type": "cloudfront"
                }
            ],
            "input-hashes": [],
            "name": "web_connectivity",
            "test-helpers": {
                "web-connectivity": "httpo://7jne2rpg5lsaqs6b.onion"
            },
            "test-helpers-alternate": {
                "web-connectivity": [
                    {
                        "address": "https://a.web-connectivity.th.ooni.io:4442",
                        "type": "https"
                    },
                    {
                        "address": "https://d2vt18apel48hw.cloudfront.net",
                        "front": "a0.awsstatic.com",
                        "type": "cloudfront"
                    }
                ]
            },
            "version": "0.0.1"
        }
    ]
}
```

After we have implemented this initial step, and we have it working both as an integration test and with unit tests, we can then make another step and integrate this with `nettests/runner.hpp`.

**Comment by bassosimone** [comment]
As an issue, this could not be 'ready for review' but only 'in progress' :)

**Comment by bassosimone** [comment]
Pull request #1016 is the place where we finish up this effort.

**Comment by bassosimone** [comment]
Enabled since #1016 has been merged into the `v0.7.x` integration branch.
